pwa/src/templates/templateParts/sourcesForm/SourcesAuthFormTemplate.tsx

```diff
@@ -24,59 +25,95 @@ export const SourcesAuthFormTemplate: React.FC<SourcesAuthFormTemplateProps & Re
   errors,
   disabled,
 }) => {
-  const { t } = useTranslation();
-
   switch (selectedAuth) {
-    case "none":
-      return <></>;
     case "apikey":
-      return (
-        <FormField>
-          <FormFieldInput>
-            <FormFieldLabel>{t("Api key")}</FormFieldLabel>
-            <Textarea {...{ register, errors, disabled }} name="apikey" validation={{ maxLength: 225 }} />
-            {errors["apikey"] && <ErrorMessage message={errors["apikey"].message} />}
-          </FormFieldInput>
-        </FormField>
-      );
+      return <ApiKeyForm {...{ register, errors, disabled }} />;
+
+    case "jwt-HS256":
+      return <JwtForm {...{ register, errors, disabled }} />;
 
     case "username-password":
-      return (
-        <>
-          <FormField>
-            <FormFieldInput>
-              <FormFieldLabel>{t("Username")}</FormFieldLabel>
-              <InputText
-                {...{ register, errors, disabled }}
-                name="username"
-                validation={{ required: true, maxLength: 225 }}
-              />
-              {errors["username"] && <ErrorMessage message={errors["username"].message} />}
-            </FormFieldInput>
-          </FormField>
-
-          <FormField>
-            <FormFieldInput>
-              <FormFieldLabel>{t("Password")}</FormFieldLabel>
-              <InputText
-                {...{ register, errors, disabled }}
-                name="password"
-                validation={{ required: true, maxLength: 225 }}
-              />
-              {errors["password"] && <ErrorMessage message={errors["password"].message} />}
-            </FormFieldInput>
-          </FormField>
-        </>
-      );
-
-    case "jwt":
-      return (
-        <FormField>
-          <FormFieldInput>
-            <FormFieldLabel>{t("JWT")}</FormFieldLabel>
-            <Textarea {...{ register, errors, disabled }} name="jwt" />
-          </FormFieldInput>
-        </FormField>
-      );
+      return <UsernamePasswordForm {...{ register, errors, disabled }} />;
+
+    case "vrijbrp-jwt":
+      return <UsernamePasswordForm {...{ register, errors, disabled }} />;
```

Comment:
Should this render a username and password form?

Reply:
According to the backend it should.